ct_optcon/examples/NLOC_generalConstrained.cpp

```diff
@@ -3,180 +3,269 @@
+              controller = nullptr)
+            : ct::core::ControlledSystem<STATE_DIM, CONTROL_DIM>(
+            controller, ct::core::SYSTEM_TYPE::SECOND_ORDER),
+              m_(m) {}
+
+    PointMass(const PointMass &other)
+            : ct::core::ControlledSystem<STATE_DIM, CONTROL_DIM>(other),
+              m_(other.m_) {}
+
+    // deep clone
+    PointMass *clone() const override { return new PointMass(*this); }
+
+    virtual ~PointMass() = default;
+
+    // compute the system dynamics
+    virtual void
+    computeControlledDynamics(const ct::core::StateVector<STATE_DIM> &state,
+                              const ct::core::Time &t,
+                              const ct::core::ControlVector<CONTROL_DIM> &control,
+                              ct::core::StateVector<STATE_DIM> &dxdt) override {
+        dxdt(0) = state(2);
+        dxdt(1) = state(3);
+        dxdt(2) = control(0) / m_;
+        dxdt(3) = control(1) / m_;
+    }
+
+private:
+    double m_; // point mass
+};
 
 /*!
  * @brief A simple 1d constraint term.
  *
  * This term implements the general inequality constraints
- * \f$ d_{lb} \leq u \cdot p^2 \leq d_{ub} \f$
- * where \f$ p \f$ denotes the position of the oscillator mass.
+ * \f$ (x - x0)^2 + (y - y0)^2 \geq r^2 \f$
+ * where \f$ r \f$ denotes the radius and \f$ (x0, y0) \f$ the position of an obstacle.
  *
- * This constraint can be thought of a position-varying bound on the control input.
- * At large oscillator deflections, the control bounds shrink
+ * This constraint encodes a behavior where a moving object avoids obstacles which
+ * are represented by a circle.
  */
-class ConstraintTerm1D : public ct::optcon::ConstraintBase<state_dim, control_dim>
-{
+
+ // class representing a general constraint that ensures obstacle avoidance
+class ObstacleConstraintSimple
+        : public ct::optcon::ConstraintBase<STATE_DIM, CONTROL_DIM> {
 public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
-    typedef ct::optcon::ConstraintBase<state_dim, control_dim> Base;
-    typedef ct::core::StateVector<state_dim> state_vector_t;
-    typedef ct::core::ControlVector<control_dim> control_vector_t;
+    typedef ct::optcon::ConstraintBase<STATE_DIM, CONTROL_DIM> Base;
+    typedef ct::core::StateVector<STATE_DIM> state_vector_t;
+    typedef ct::core::ControlVector<CONTROL_DIM> control_vector_t;
+    typedef Eigen::Matrix<double, 2, 1> Vector2s;
 
-    //! constructor with hard-coded constraint boundaries.
-    ConstraintTerm1D()
-    {
+    ObstacleConstraintSimple(const Vector2s x0, const double r) : x0_(x0), r_(r) {
         Base::lb_.resize(1);
         Base::ub_.resize(1);
-        Base::lb_.setConstant(-1.0);
-        Base::ub_.setConstant(1.0);
+        Base::lb_.setConstant(0.0);
+        Base::ub_.setConstant(std::numeric_limits<double>::max());
+    }
+
+    ObstacleConstraintSimple(const ObstacleConstraintSimple &other)
+            : Base(other), x0_(other.x0_) {
+        Base::lb_.resize(1);
```

Comment:
this initializer list should not be required, that should be taken care of by the copy-constructor of the base class (but maybe you'd want to confirm that).

Reply:
Confirmed, thank you!